src/components/chown_component.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+# Copyright 2025 Ubuntu
+# See LICENSE file for licensing details.
+
+import logging
+from typing import Optional
+
+from charmed_kubeflow_chisme.components.component import Component
+from charmed_kubeflow_chisme.exceptions import ErrorWithStatus
+from ops import ActiveStatus, CharmBase, StatusBase, WaitingStatus
+
+logger = logging.getLogger(__name__)
+
+
+class ChownMountedStorageComponent(Component):
+    """A component that ensures mounted storage is owned by the user that runs the service.
+
+    Args:
+        charm(CharmBase): this charm
+        name(str): name of the component
+        storage_path(str): the path where the storage is mounted and is desired to change owners.
+        workload_user(str): the user that runs the service (workload).
+    """
+
+    def __init__(
+        self,
+        charm: CharmBase,
+        name: str,
+        storage_path: Optional[str] = "/data",
+        workload_user: Optional[str] = "_daemon_",
```

Comment:
suggestion (non-blocking): Maybe it makes sense to not specify default arguments here, so that we have to give them in `charm.py`, and change them from where we create the component if the workload itself changes.

```suggestion
        storage_path: Optional[str] = "/data",
        workload_user: Optional[str] = "_daemon_",
```